### DISCUSSION TITLE
logging format / use of f-strings

### DISCUSSION THREAD

**Discussion by mrosseel** [WimDeMeester/eFinder#43] [General]
﻿Just a quick note on logging:

While merging I saw these statements:
==> logging.info("Align status reply ", p[0:3])

being replaced by these statements:
==> logging.info("%s %s" % ("Align status reply ", p[0:3]))

I'm guessing because of errors regarding logging.info only taking one argument?
In that case, a better fix is using the f-strings of python (note the f letter preceding the string):

==> logging.info(f"Align status reply {p[0:3]}")

**Comment by WimDeMeester** [comment]
Thanks! I'll make a pull request to fix this.  

A question on the same topic: is it easy to log everything in an easy way to a file (with the date in the filename)?

**Reply by mrosseel** [reply]
please don't, I have enough merge conflicts as it is :)

Yes it's pretty easy, I have the code somewhere

```
filehandler = f"{datadir}vastlog-{datenow:%Y%M%d-%H_%M_%S}.log"
fh = logging.FileHandler(filehandler)
fh.setLevel(logging.INFO)
# add the handlers to the logger
logger.addHandler(fh)
```

**Reply by WimDeMeester** [reply]
> please don't, I have enough merge conflicts as it is :)

So I guess you will do the changes in your upcoming pull request.

**Reply by WimDeMeester** [reply]
> Yes it's pretty easy

Can we add a command line option for this?  I think it would nice (certainly when doing the first tests outside) to have everything logged in a file.  
Can I do this, or do you prefer to do it yourself, so that I don't introduce more merge conflicts?

**Comment by mrosseel** [comment]
yes these changes should be included, I'll add it in my branch.
